src/graphql/resolvers/address-resolver.ts

```diff
@@ -70,4 +70,14 @@ export class AddressResolver {
       }
     });
   }
+
+  @FieldResolver()
+  async used(@Root() root: Address, @Ctx() context: GraphQLContextWithArgs<ContextArgs>) {
+    return await context.repository.boxes.getAddressBoxCount(root.address) > 0;
+  }
+
+  @FieldResolver()
+  async boxesCount(@Root() root: Address, @Ctx() context: GraphQLContextWithArgs<ContextArgs>) {
+    return await context.repository.boxes.getAddressBoxCount(root.address);
+  }
```

Comment:
1. Addresses can be queried in batches of 20, so, for 20 addresses, if `used` and `boxesCount` are selected in the same query, 40 database roundtrips will be necessary just to resolve those. We can optimize it by resolving them on `addresses()` method using `isFieldSelected()` function to check for selections and batch querying boxes just like `BoxRepository.sum()` does.

  2. `atHeight` param should be taken into account here.